corehq/apps/cloudcare/esaccessors.py

```diff
@@ -39,28 +39,13 @@ def login_as_user_query(
         user_es = user_es.location(list(loc_ids))
 
     if _limit_login_as(couch_user, domain):
-        user_filters = [login_as_user_filter(couch_user.username)]
+        usernames = [couch_user.username]
         if couch_user.has_permission(domain, 'access_default_login_as_user'):
-            user_filters.append(login_as_user_filter('default'))
-        user_es = user_es.filter(
-            queries.nested(
-                'user_data_es',
-                filters.OR(
-                    *user_filters
-                )
-            )
-        )
+            usernames.append('default')
+        user_es = user_es.user_data('login_as_user', usernames)
```

Comment:
Should this be domain-scoped?  Like, could I set `login_as_user` on another domain to the username of someone I shouldn't be able to access on the main domain, and then access them anyways?

Reply:
That's a good point, I guess that behaviour is indeed possible now, but we can fix that now (that we have nested fields for domain_memberships)?

Reply:
Ah right yeah that makes sense